src/object/cli_obj.c

```diff
@@ -2712,6 +2719,9 @@ shard_auxi_set_param(struct shard_auxi_args *shard_arg, uint32_t map_ver,
 		     uint32_t shard, uint32_t tgt_id, struct dtx_epoch *epoch,
 		     uint16_t ec_tgt_idx)
 {
+	/* Reset @enqueue_id if target changed */
+	if (shard_arg->target != tgt_id)
+		shard_arg->enqueue_id = 0;
```

Comment:
the query key operation is not affected by this change, you may also need to change queue_shard_query_key_task().
Does it work if use HLC as the enqueue_id at server-side? by that need not such change at client-side.

Reply:
@NiuYawei and i discussed a bit that way, we think it better to reset enqueued ID as it is a new request. (send a TLC to new target means new RPC could jump a queue).

Let me fix shard_query_key_task() case thanks!